OpenRA.Mods.Common/Traits/Passenger.cs

```diff
@@ -183,5 +183,14 @@ public void Unreserve(Actor self)
 			ReservedCargo.UnreserveSpace(self);
 			ReservedCargo = null;
 		}
+
+		void INotifyKilled.Killed(Actor self, AttackInfo e)
+		{
+			if (Transport == null || Transport.IsDead)
+				return;
+
+			var c = Transport.Trait<Cargo>();
```

Comment:
This can be simplified to `Transport.Trait<Cargo>().Unload(Transport, self);`

Reply:
It cannot be as the Transport may be null when you surrender

Reply:
although the IsDead check is unnecessary

Reply:
I mean just the `var c = ` bit, which you've now fixed.